torch/_inductor/virtualized.py

```diff
@@ -91,6 +90,29 @@ def inner(*args):
             setattr(cls, name, make_handler(format_string))
 
 
+class KernelFormatterHandler:
+    def __init__(self, parent_handler):
+        self.parent_handler = parent_handler
+        self.output = IndentedBuffer()
+        self.var_counter = itertools.count()
+
+    def __getattr__(self, name):
+        def inner(*args, **kwargs):
+            line = getattr(self.parent_handler, name)(*args, **kwargs)
+            if name == "indirect_indexing":
+                return line
+            # replace line with a new variable name
+            varname = f"tmp{next(self.var_counter)}"
+            self.output.writeline(f"{varname} = {line}")
+            return varname
+
+        return inner
+
+    def getvalue(self, result):
+        self.output.writeline(f"return {result}")
+        return self.output.getvalue()
```

Comment:
Without the expression tree it's no longer possible to do dead-code elimination.